app/assets/javascripts/initializers/initializePodcastPlayback.js

```diff
@@ -272,52 +341,128 @@ function initializePodcastPlayback() {
     sec = sec >= 10 ? sec : '0' + sec;
     return min + ':' + sec;
   }
-  
 
   function terminatePodcastBar(audio) {
-    audio.removeEventListener('timeupdate', updateProgressListener, false);
+    audio.removeEventListener('timeupdate', updateProgressListener(audio), false);
     getById('audiocontent').innerHTML = '';
     stopRotatingActivePodcastIfExist();
-    setMediaState(audio);
+    if (isNativeIOS()) {
+      sendPodcastMessage('terminate');
+    }
   }
 
-  function setMediaState(audio) {
-    var date = new Date();
-    var timeNumber = date.getTime();
+  function saveMediaState(state) {
+    var currentState = state || currentAudioState();
+    var newState = newAudioState();
+    newState.currentTime = currentState.currentTime;
+    newState.playing = currentState.playing;
+    newState.muted = currentState.muted;
+    newState.volume = currentState.volume;
+    newState.duration = currentState.duration;
+    localStorage.setItem('media_playback_state_v2', JSON.stringify(newState));
+  }
+
+  function getMediaState() {
+    var currentState = currentAudioState();
+    document.getElementById('audiocontent').innerHTML = currentState.html;
+    var audio = getById('audio');
+    if (audio == undefined) {
+      return;
+    }
+    if (!isNativeIOS()) {
+      audio.currentTime = currentState.currentTime || 0;
```

Comment:
I believe this is the troublemaker that causes the playback glitch when browsing different pages

Reply:
I've tried to avoid setting `audio.currentTime` if the audio is currently playing (`!audio.paused`) but the audio HTML element is kind of behaving ina weird way (always shows that paused value as `false`). Still looking into this 🧐

Reply:
Resolved thanks to help from Ben  🎉. New commits to be shared soon